### PR TITLE
chore: update to rust-analyzer 2022-12-26

### PR DESCRIPTION
**Pull request by onlyhavecans**
Catching up to all the new features!

### CODE REVIEW

**Comment by rwols** [comment]
Thanks! We should wait with merging this until https://github.com/sublimelsp/LSP/issues/2152 is released.

**Comment by rchl** [comment]
Just released new LSP version.

**Comment by rchl** [comment]
This version should also support `rust-analyzer.inlayHints.locationLinks` but I don't see it in the added settings.

File: LSP-rust-analyzer.sublime-settings [hunk 1]
```diff
@@ -59,10 +76,25 @@
...
+		// - "workspace": run checks for workspaces in the corresponding workspaces' root directories.
+		// This falls back to "root" if rust-analyzer.cargo.checkOnSave.invocationStrategy is set to once.
+		// - "root": run checks in the project’s root directory.
+		// This config only has an effect when rust-analyzer.cargo.buildScripts.overrideCommand is set.
+		"rust-analyzer.checkOnSave.invocationLocation": "workspace",
+		// Specifies the invocation strategy to use when running the checkOnSave command.
+		// If per_workspace is set, the command will be executed for each workspace.
+		// If once is set, the command will be executed once.
+		// This config only has an effect when rust-analyzer.cargo.buildScripts.overrideCommand is set.
+
```
Comment (rchl) [inline LSP-rust-analyzer.sublime-settings:96]: can drop this newline :)

**Comment by onlyhavecans** [comment]
Sorry about the missed setting and whitespace 🙏🏻 
I fixed both of them

**Review by rchl — APPROVED** [review]

**Pull request merged**